SeqVision/SeqVision.py

```diff
@@ -0,0 +1,143 @@
+from typing import List, Union, Callable, Dict, Tuple
+import os
+from modules.fastq_tool_module import (
+    calculate_gc_content,
+    filter_reads_by_gc,
+    filter_length_bounds,
+    filter_quality_threshold,
+)
+from modules.dna_rna_tools_module import (
+    is_rna,
+    is_dna,
+    complement,
+    reverse,
+    transcribe,
+    reverse_complement,
+    which_palindrome,
+)
+
+
+def filter_fastq(
+    input_fastq: str,
+    output_fastq: str,
+    gc_bounds: Union[Tuple[int, int], int, float] = (0, 100),
+    length_bounds: Union[Tuple[int, int], int] = (0, 2**32),
+    quality_threshold: Union[int, float] = 0,
+) -> Dict[str, Tuple[str, str]]:
+    """
+    Filters FASTQ reads by GC content, sequence length, and quality threshold.
+
+    Parameters:
+    ----------
+    input_fastq : str
+        Name of the FASTQ file located in the "data: folder.
+
+    output_fastq : str
+        Name of the output FASTQ file 
+        that will be located in the "filtered" folder.
+
+    gc_bounds : Union[Tuple[int, int], int, float], optional
+        GC content interval (in percent) for filtering.
+        If a single number is provided, it is considered the upper bound,
+        and the lower bound is set to 0.
+        Defaults to (0, 100), i.e., all reads are preserved.
+
+    length_bounds : Union[Tuple[int, int], int], optional
+        Sequence length interval for filtering.
+        If a single number is provided, it is considered the upper bound,
+        and the lower bound is set to 0.
+        Defaults to (0, 2**32), i.e., all reads are preserved.
+
+    quality_threshold : Union[int, float], optional
+        Average quality threshold for filtering.
+        Defaults to 0, i.e., all reads are preserved.
+
+    Returns:
+    --------
+    Dict[str, Tuple[str, str]]
+        Filtered dictionary of reads that meet the specified conditions
+        for GC content, length, and quality.
+    """
+    if isinstance(gc_bounds, (int, float)):
+        gc_bounds = (0, gc_bounds)
+    if isinstance(length_bounds, int):
+        length_bounds = (0, length_bounds)
+    file_path = os.path.join("data", input_fastq)
+    output_file_path = os.path.join("filtered", output_fastq)
+    with open(file_path, "r") as fastq_file:
+        with open(output_file_path, "a") as output_file:
+            while True:
+                id_line = fastq_file.readline().strip()
+                if not id_line:
+                    break
+                seq_line = fastq_file.readline().strip()
+                id2_line = fastq_file.readline().strip()
+                quality_line = fastq_file.readline().strip()
+                if not (id_line.startswith("@SR") 
+                        and id2_line.startswith("+SR")):
```

Comment:
Мне кажется, можно было бы оставить только @ и +, чтобы функция могла работать с любым fastq файлом
```suggestion
                if not (id_line.startswith("@") 
                        and id2_line.startswith("+")):
```

Reply:
Правильно!